src/phoenix/server/api/types/Project.py

```diff
@@ -45,6 +57,55 @@ def latency_ms_p50(self) -> Optional[float]:
     def latency_ms_p99(self) -> Optional[float]:
         return self.project.root_span_latency_ms_quantiles(0.99)
 
+    @strawberry.field
+    def spans(
+        self,
+        info: Info[Context, None],
+        time_range: Optional[TimeRange] = UNSET,
+        trace_ids: Optional[List[ID]] = UNSET,
+        first: Optional[int] = 50,
+        last: Optional[int] = UNSET,
+        after: Optional[Cursor] = UNSET,
+        before: Optional[Cursor] = UNSET,
+        sort: Optional[SpanSort] = UNSET,
+        root_spans_only: Optional[bool] = UNSET,
+        filter_condition: Optional[str] = UNSET,
+    ) -> Connection[Span]:
+        args = ConnectionArgs(
+            first=first,
+            after=after if isinstance(after, Cursor) else None,
+            last=last,
+            before=before if isinstance(before, Cursor) else None,
+        )
+        if not (traces := info.context.traces) or not (
+            project := traces.get_project(DEFAULT_PROJECT_NAME)
+        ):
+            return connection_from_list(data=[], args=args)
```

Comment:
you could add a `__bool__` method to Project so that you can exit early
```suggestion
        project = self.project
```

Reply:
or 
```suggestion
        if not (project := self.project).span_count:
            return connection_from_list(data=[], args=args)
```

Reply:
I think long-term, projects could be used for more than just spans, so let's not make the `__bool__`.